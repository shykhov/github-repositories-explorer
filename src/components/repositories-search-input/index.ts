export * from './repositories-search-input';
