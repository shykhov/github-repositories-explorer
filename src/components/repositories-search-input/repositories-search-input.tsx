import React, { ChangeEvent } from 'react';
import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import InputAdornment from '@material-ui/core/InputAdornment';
import CircularProgress from '@material-ui/core/CircularProgress';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: '2px 4px',
      display: 'flex',
      alignItems: 'center',
      width: 400,
    },
    input: {
      marginLeft: theme.spacing(1),
      flex: 1,
    },
    adornment: {
      minWidth: 20,
    },
    iconButton: {
      padding: 10,
    },
    divider: {
      height: 28,
      margin: 4,
    },
  }),
);

interface Props {
  handleSearchInputChange(event: ChangeEvent<HTMLInputElement>): void;
  loading: boolean;
  value: string | undefined;
}

export const RepositoriesSearchInput = (props: Props) => {
  const { handleSearchInputChange, loading, value } = props;
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <InputBase
        value={value}
        onChange={handleSearchInputChange}
        className={classes.input}
        placeholder="Search for repository name"
        startAdornment={
          <InputAdornment className={classes.adornment} position="start">
            {loading && <CircularProgress size={20} />}
          </InputAdornment>
        }
        inputProps={{ 'aria-label': 'search github repository' }}
      />
      <IconButton className={classes.iconButton} aria-label="search">
        <SearchIcon />
      </IconButton>
    </Paper>
  );
};
