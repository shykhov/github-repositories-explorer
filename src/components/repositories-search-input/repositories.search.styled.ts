import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import InputAdornment from '@material-ui/core/InputAdornment';
import IconButton from '@material-ui/core/IconButton';

export const StyledPaper = styled(Paper)`
  padding: 2px 4px;
  display: flex;
  align-items: center;
`;

export const StyledInputBase = styled(InputBase)`
  margin-left: 41px;
  font-family: 'Lato', sans-serif;
  flex: 1;
`;

export const StyledInputAdornment = styled(InputAdornment)`
  max-width: 20px;
`;

export const StyledIconButton = styled(IconButton)`
  padding: 10px;
`;
