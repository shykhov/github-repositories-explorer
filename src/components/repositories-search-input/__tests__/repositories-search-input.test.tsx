import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { RepositoriesSearchInput } from '../repositories-search-input';

describe('<RepositoriesSearchInput />', () => {
  let renderedComponent: ShallowWrapper;
  let propsMock;

  beforeEach(() => {
    propsMock = {
      handleSearchInputChange: jest.fn(),
      debouncedSubmitInputSearch: { flush: jest.fn() },
      loading: false,
      value: 'some',
    };

    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    renderedComponent = shallow(<RepositoriesSearchInput {...propsMock} />);
  });

  it('should render RepositoriesSearchInput', () => {
    expect(renderedComponent.dive()).toMatchSnapshot();
  });
});
