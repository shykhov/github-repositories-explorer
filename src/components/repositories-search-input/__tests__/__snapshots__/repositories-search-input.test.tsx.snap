// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoriesSearchInput /> should render RepositoriesSearchInput 1`] = `
<WithStyles(ForwardRef(Paper))
  className="sc-AxjAm ddhvir"
  elevation={10}
>
  <Styled(WithStyles(ForwardRef(InputBase)))
    inputProps={
      Object {
        "aria-label": "search github repository",
      }
    }
    onChange={[MockFunction]}
    onKeyDown={[Function]}
    placeholder="Search for repository name"
    startAdornment={
      <ForwardRef(Styled(WithStyles(ForwardRef(InputAdornment))))
        position="start"
      />
    }
    value="some"
  />
  <Styled(WithStyles(ForwardRef(IconButton)))
    aria-label="search"
    onClick={[Function]}
  >
    <SearchIcon />
  </Styled(WithStyles(ForwardRef(IconButton)))>
</WithStyles(ForwardRef(Paper))>
`;
